cookbook/core/control_flow/dynamics.py

```diff
@@ -1,56 +1,119 @@
 """
-Write a dynamic task
-------------------------------------------
+Dynamic Workflow
+----------------
 
-Please make sure you understand the difference between a task and a workflow - feel free to revisit the earlier
-discussions and examples of those concepts. The key thing to note is that the body of tasks run at run time and the
-body of workflows run at compile (aka registration) time. A dynamic task is a mix between the two - it's effectively
-a workflow that is compiled at run time.
+A workflow is typically static where the directed acyclic graph's (DAG) structure is known well ahead of time. However, scenarios could exist where an "unknown" variable only known at run-time could determine the proper DAG structure.
+
+In such cases, dynamic workflows can be used. Here's a code example that counts the common characters between any two strings.
+
+Inputs: s1 = "Pear", s2 = "Earth"
+
+Output: 3
 
 """
+
+# %%
+# Let's first import all the required libraries.
 import typing
 
 from flytekit import dynamic, task, workflow
 
+# %%
+# Next, we write a task that returns the index of a character (A-Z/a-z is equivalent to 0 to 25).
+@task
+def return_index(character: str) -> int:
+    """
+    Computes the character index (which needs to fit into the 26 characters list)"""
+    if character.islower():
+        return ord(character) - ord("a")
+    else:
+        return ord(character) - ord("A")
 
+
+# %%
+# We now write a task that prepares the 26 characters list by populating the frequency of every character.
 @task
-def t1(a: int) -> str:
-    a = a + 2
-    return "world-" + str(a)
+def prepare_list(freq_list: typing.List[int], list_index: int) -> typing.List[int]:
```

Comment:
consider `update_list` as a more apt name for this function

Reply:
Updated the same.